app/src/main/java/com/example/cinemapp/ui/main/MainActivity.kt

```diff
@@ -22,12 +43,158 @@ class MainActivity : AppCompatActivity() {
         setupNavigation()
     }
 
+    override fun onStart() {
+        super.onStart()
+        viewModel.getAccountDetail(false)
+        setupOnClickListeners()
+        observeEvents()
+
+        observeFlowSafely(viewModel.state) {
+            it.accountDetails?.let { accountDetails ->
+                setupAvatar(accountDetails, binding.tvInitial, binding.ivAvatar)
+            }
+        }
+
+    }
+
     private fun setupNavigation() {
         val navHostFragment =
             supportFragmentManager.findFragmentById(binding.navHostFragment.id) as NavHostFragment
         navController = navHostFragment.navController
         binding.bottomNavigationView.setupWithNavController(navController)
     }
 
+    private fun setupAvatar(
+        accountDetails: AccountDetails?,
+        tvInitial: TextView,
+        ivAvatar: ImageView,
+    ) {
+        accountDetails?.let { details ->
+            if (details.avatar.isEmpty()) {
+                tvInitial.text =
+                    (if (details.name.isEmpty()) details.username else details.name)
+                        .substring(0, 1)
+                ivAvatar.setBackgroundColor(
+                    ContextCompat.getColor(this@MainActivity, R.color.md_theme_onPrimary)
+                )
+                ivAvatar.setImageDrawable(null)
+
+            } else {
+                tvInitial.text = ""
+                ivAvatar.setBackgroundColor(
+                    ContextCompat.getColor(
+                        this@MainActivity,
+                        R.color.md_theme_primary
+                    )
+                )
+                loadImage(
+                    accountDetails.avatar,
+                    ivAvatar,
+                    this@MainActivity,
+                    R.drawable.vic_profile
+                )
+                ivAvatar.scaleX = 1f
+                ivAvatar.scaleY = 1f
+                ivAvatar.imageTintList = null
+            }
+        } ?: run {
+            tvInitial.text = ""
+            ivAvatar.setBackgroundColor(
+                ContextCompat.getColor(this@MainActivity, R.color.md_theme_primary)
+            )
+            ivAvatar.setImageResource(R.drawable.vic_profile)
+            ivAvatar.scaleX = 1.2f
+            ivAvatar.scaleY = 1.2f
+            ivAvatar.setColorFilter(
+                ContextCompat.getColor(
+                    this@MainActivity,
+                    R.color.md_theme_onPrimary
+                )
+            )
+        }
+    }
+
+    private fun setupOnClickListeners() {
+        binding.rlAvatar.setOnClickListener {
+            viewModel.getAccountDetail(true)
```

Comment:
Viewmodel should know if the dialog is opened from the state, also this is little unreadable, without context what does 'true' mean?

Reply:
So, there should be a field in the state that indicates if the dialog is opened?
As for the context on what true means, the name of the parameter explains it.